proposals/web-remote-pinning-service.md

```diff
@@ -0,0 +1,160 @@
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+go-ipfs 0.8.0 and [IPFS HTTP Client][] have shipped support for vendor-agnostic [remote pinning services][]. [Pinata][] has shipped support and [Textile is commited][] to doing it as well. This enables users to persist and ensure their data is available without having to invest into own infrastructure, nor risking a vendor lock-in. 
+
+Support has not been implemented in js-ipfs making this feature unavailable for web nodes, so teams building web3 products will need to contiune providing persistence through a backend as opposed to letting their users choose where their data gets persisted.
+
+By implementing `pin remote` commands in js-ipfs we can enable ecosystem of web applications in which product users are empowered to choose where their data is persisted with almost no effort on teams building them.
+
+[remote pinning services]:https://ipfs.github.io/pinning-services-api-spec/
+[pinata]:https://pinata.cloud
+[Textile is commited]:https://twitter.com/textileio/status/1363896959073804288?s=20
+[IPFS HTTP Client]:https://www.npmjs.com/package/ipfs-http-client
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+- Teams building products would like an ability of not having to build persistence layer.
+- Pinning services will deploy support for this API.
+- Dapp / NFT developers see value in empowering users with ability to use custom / third party pinning services.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+App developer will have a choice to not build their own storage, instead allowing a user connect app with their Pinata, Textile, or any other pinning service account _(with a flow similar to login with twitter/facebook/google)_ to register remote pinning service into app embedded IPFS node.
+
+This would allow application to pin data at  pinning service of developer's or user's choosing, increasing user agency over their data, and simplifying long term storage story for web apps, dapps  and NFTs.
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+- Web3 is all about making users in charge of their data. This makes it possible out of the box without any additional effort.
+- Enables a class of web3 applications that run all of the business logic at the edge to be backend free (app is deployed as a static bundle of html+js, users bring their data storage through pinning services).
```

Comment:
how would this handle pinning service API tokens?

Reply:
> how would this handle pinning service API tokens?

Implementation-vise application will obtain service endpoint+token from the user and add it to embedded js-ipfs node which will persist it in the browser storage. That said I hope this would prompt things like sign-in with pinata/textile/etc.. style interaction where users has to do nothing other than click a link which will take them to the desired service and redirect back passing all the endpoint+token information through query arguments.